### PR TITLE
7806 missing collateral

### LINKED ISSUE #7806: ATOM assets missing following liquidation and reconstitution
## Describe the bug
In validation testing the second half of #7784 - impact to reserve metrics, two issues were observed:
1. Total amounts distributed were less than what is going in (assets missing) - 14.99 (returned to vault) + 14.99 (returned to vault) + 10.01 (bid payout) + 3.575 (bid payout) + 1.107 (ATOM in reserve) vs 45 ATOM
2. vaults seem to have too much and no longer match expectations (each reconstituted vault was expected to receive 14.889 ATOM / 100.5 IST and 14.896 ATOM / 103.51 IST back after reconstitution but had 14.99 ATOM /100.5IST and 14.99 ATOM/ 103.51 IST respectively) 

## To Reproduce
1. Using 12.34 price set up vaults per Flow 2b/reconstitution and 2x bid per https://docs.google.com/document/d/1bXnQ_o3wwfnu-uhX2KSb6ZXB0dU6CsAsGBWm1DXgmE0/edit?usp=sharing 
4. trigger liquidaion with 9.99 pushPrice
5. Check reserve metrics before auction starts
6. Observe liquidation and auction take place
7. Review reconstituted vault information, bid payouts and post-auction reserve metrics

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Additional context
Reserve metrics before auction:

root@3ac654d9e579:/usr/src/agoric-sdk# agoric follow -lF :published.reserve.metrics
{
  allocations: {},
  shortfallBalance: {
    brand: slotToVal("board0257","Alleged: IST brand"),
    value: 0n,
  },
  totalFeeBurned: {
    brand: slotToVal("board0257"),
    value: 0n,
  },
  totalFeeMinted: {
    brand: slotToVal("board0257"),
    value: 0n,

Reserve metrics after:

Reserve after:

  allocations: {
    ATOM: {
      brand: slotToVal("board05557","Alleged: ATOM brand"),
      value: 1107181n, (1.107 IST)
    },
    Fee: {
      brand: slotToVal("board0257","Alleged: IST brand"),
      value: 2135000n, (2.1 IST)
    },
  },
  shortfallBalance: {
    brand: slotToVal("board0257"),
    value: 5525000n, (5.525)
  },
  totalFeeBurned: {
    brand: slotToVal("board0257"),
    value: 0n,
  },
  totalFeeMinted: {
    brand: slotToVal("board0257"),
    value: 0n,

## Screenshots
Pasted Graphic 10
Pasted Graphic 11

### LINKED ISSUE #7837: Unexpected excess collateral amounts distributed to happy path liquidated vaults
## Describe the bug
When testing happy path liquidation scenario following #7779 during pre-RC cutting testing  - I noticed a slight discrepancy in the excess collateral each vault received vs what was expected. I one case - it was higher than expected and then lower than expected for other two vaults.

The delta is not quite the penalty (thought it might be penalty-related):
Expected - Actual = Delta
Vault 1: 3.373-3.52 = +0.147
Vault 2: 3.024-3.18 = -0.156
Vault 3: 2.791-2.64 = -0.151

## To Reproduce
1. Configure vaults and bids per happy path liquidation scenario
2. Trigger liquidation by setting price to 9.99
3. Await for auction to complete and for liquidated vaults status to update to show excess collateral received

## Expected behavior
Expected to receive V1: 3.373 V2: 3.024 V3: 2.791 per liquidation model https://docs.google.com/document/d/1bXnQ_o3wwfnu-uhX2KSb6ZXB0dU6CsAsGBWm1DXgmE0/edit?usp=sharing

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Screenshots
If applicable, add screenshots to help explain your problem, especially for UI interactions.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #7806
fixes: #7837

## Description

Fixes a bug detected in manual testing. The VaultManager wasn't distributing all the remaining proceeds [in flow 2b](#7123) to the reserve. This fix sends everything remaining in the seat at the end of distribution to the reserves, and logs any discrepancies. That should sweep any leftovers to the reserve when it is first run.

In adding enough details to the tests to make them reviewable, I understood the issue in #7837, so that's fixed here as well. In `distributeProceeds`, the penalties weren't being subtracted from each vaultHolder's allowable refund.

### Security Considerations

Highlights the importance of ensuring all allocations are accounted for when reallocation is done.

### Scaling Considerations

N/A

### Documentation Considerations

None

### Testing Considerations

The fact that we already had a test for this scenario and a thorough analysis made it straightforward to compare results with expected.

### Release plan

This PR will make a change in behavior in the Vaults, but not one that requires any other code to change in response. I'll still mark it as draft so we can discuss how to stage it.

### CODE REVIEW

File: packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js [hunk 1]
```diff
@@ -2398,7 +2398,7 @@ test('Bug 7346 excess collateral to holder', async t => {
   t.false(await E(bidderSeat3).hasExited());
   await E(bidderSeat3).tryExit();
   t.true(await E(bidderSeat3).hasExited());
-  await assertBidderPayout(t, bidderSeat3, run, 10_460n, aeth, 0n);
+  await assertBidderPayout(t, bidderSeat3, run, 10_460n, aeth, 16432n);
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js]: I'm having a hard time validating that this value is correct. Can you name some of these values and express them in terms of each other?

**Comment by Chris-Hibbert** [comment]
@rowgraus, @otoole-brendan  The results here now match the scenario (now that I've corrected the spreadsheet.)


vault | collateral | actual debt | debt as collateral | penalty as collateral | max return
-|-|-|-|-|-
vault 1 | 15000000 | 100.5M | 11,474,253 | 100,601 | 3,425,146
vault 2 |  15000000 | 103.515M | 11,818,481 | 103,619 | 3.077,900
vault 3 |  15000000 | 105.525M| 12,047,966| 105,631 | 2.846,403

File: packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js]: Yes. Take another look. I multiplied everything out to match the numbers in @rowgraus' scenario, and added a lot of details.

**Comment by turadg** [comment]
if https://github.com/Agoric/agoric-sdk/pull/7804/ gets into master before this, 26772dd9492bd5b61cbaf10a1fc45f481d62370c will update the test to match these numbers

**Comment by Chris-Hibbert** [comment]
@turadg I suspect I sent the previous response before all the relevant changes had been pushed. This is now complete and up-to-date.  PTAL.

File: packages/inter-protocol/src/vaultFactory/vaultManager.js [hunk 2]
```diff
@@ -917,12 +922,29 @@ export const prepareVaultManagerKit = (
...
+            facets.helper.sendToReserve(collateralInLiqSeat, liqSeat);
+
+            if (
+              !AmountMath.isEqual(
+                collateralInLiqSeat,
+                AmountMath.add(collatRemaining, totalPenalty),
+              )
+            ) {
+              console.error(
+                `🚨 Excess collateral remaining sent to reserve. Expected ${q(
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/vaultManager.js:947]: good call checking but not throwing

File: packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js [hunk 3]
```diff
@@ -2301,11 +2326,12 @@ test('Bug 7346 excess collateral to holder', async t => {
 
   await aethVaultMetrics.assertChange({
     numActiveVaults: 2,
-    totalCollateral: { value: 30_000n },
-    totalDebt: { value: 204_015n },
+    totalCollateral: { value: 30_000_000n },
+    totalDebt: { value: aliceDebt + bobDebt },
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js:2333]: 🙏

**Review by turadg — APPROVED** [review]
Thanks for naming most of the test values.

Code change and test coverage LGTM.

Let's confer on when to merge.

**Pull request merged**